charts/opentelemetry-collector/templates/_config.tpl

```diff
@@ -34,28 +34,40 @@ Merge user supplied top-level (not particular to standalone or agent) config int
 {{- .Values.config | toYaml }}
 {{- end }}
 
-
 {{/*
-Build config file for agent OpenTelemetry Collector
+Build config file for daemonset OpenTelemetry Collector
 */}}
-{{- define "opentelemetry-collector.agentCollectorConfig" -}}
-{{- $values := deepCopy .Values.agentCollector | mustMergeOverwrite (deepCopy .Values)  }}
+{{- define "opentelemetry-collector.daemonsetConfig" -}}
+{{- $values := deepCopy .Values.agentCollector | mustMergeOverwrite (deepCopy .Values) }}
+{{- if eq .Values.mode "daemonset" }}
+{{- $values =  deepCopy .Values }}
```

Comment:
I thought this too, but I was running into a strange issue where if I used := inside the if statement block it wouldn't actually modify the value of the variable outside the if statement block. Using = would change the variable so that the new value was used outside the if statement block.

Reply:
I've not run into this kind of issue before but this is also the first time I've done this type of "programming-like" variable override.

Reply:
Ok sounds good. Please verify that it's working as expected in few modern helm versions

Reply:
Sure, I'll check the current and last 3 stable versions. I think this was the SO post I found to help through the issue. https://stackoverflow.com/questions/57600772/is-it-possible-to-define-variables-use-if-else-condition-in-helm-chart

Reply:
Nice. Thanks for posting this

Reply:
Testing using `generate-charts` with versions 3.7.0, 3.7.1, 3.7.1, 3.8.0, and 3.8.1.  `if` conditions worked as expected.